lib/OpenQA/WebAPI/Controller/Step.pm

```diff
@@ -371,26 +372,27 @@ sub _extended_needle_info {
 
 sub src {
     my ($self) = @_;
-
     return $self->reply->not_found unless $self->init();
 
     my $job    = $self->stash('job');
     my $module = $self->stash('module');
 
+    if (my $casedir = $job->settings->find({key => 'CASEDIR'})) {
```

Comment:
It seems there's no unique constraint for the column `key` so you better use `search` and deal explicitly with the fact that there can be more than just one row (e.g. order by ID and always take the last). Otherwise we might get warnings from DBIx.

Reply:
how about `first` instead?

Reply:
… I meant `single`